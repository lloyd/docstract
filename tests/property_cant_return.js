/** A property cannot return data */

/** @prop foo
 * @returns {string} whatever, this is silly. */
