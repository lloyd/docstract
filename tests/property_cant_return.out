{
  "args": [
    "@return not allowed in @property block at line 3"
  ], 
  "exception_type": "<type 'exceptions.RuntimeError'>", 
  "stack": [
    [
      "<module>", 
      "got = ds.extractFromFile(os.path.join(testDir, test + \".js\"))"
    ], 
    [
      "extractFromFile", 
      "data = self.extract(contents)"
    ], 
    [
      "extract", 
      "self._analyzeBlock(block, context, firstBlock, data, lineStart, line)"
    ], 
    [
      "_analyzeBlock", 
      "(tag, parseData['blockHandler'].tagName))"
    ]
  ]
}
